ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/FSharpResolveUtil.fs

```diff
@@ -81,3 +82,13 @@ let mayShadowPartially (newExpr: ITreeNode) (data: ElementProblemAnalyzerData) (
     |> Seq.collect (fun element -> opens.GetValuesSafe(element.ShortName))
     |> Seq.toArray
     |> OpenScope.inAnyScope newExpr
+
+let isPredefinedFunctionShadowed (context: ITreeNode) name opName =
+    let checkerService = context.GetContainingFile().As<IFSharpFile>().CheckerService
+    match checkerService.ResolveNameAtLocation(context, [name], opName) with
+    | Some symbolUse ->
+        let symbol = symbolUse.Symbol
+        match predefinedFunctionTypes.TryGetValue(symbol.DisplayName) with
+        | true, fullName -> (fullName.FullName + "." + symbol.DisplayName) <> symbol.FullName
```

Comment:
About `predefinedFunctionTypes.TryGetValue(symbol.DisplayName)`:
`symbol.DisplayName` may be wrong for `predefinedFunctionTypes` here due to some of the names get escaped. Since you're already `name` as a parameter, what's the purpose of calculating `DisplayName` from the symbol?

About `fullName.FullName + "." + symbol.DisplayName`:
Why calculate `symbol.DisplayName` from the symbol once again? Why the full name is even needed here?
Could it be something like this instead?
```fsharp
match predefinedFunctionTypes.TryGetValue(name), symbolUse.Symbol.DeclaringEntity with
| (true, typeName), Some fcsEntity -> typeName.FullName <> fcsEntity.FullName
| _ -> false
```